### DISCUSSION TITLE
Do I need to run the app all the time?

### DISCUSSION THREAD

**Discussion by pauborrasblanco** [flobz/psa_car_controller#607] [Q&A] [answered]
This may be the dumbest question. I am far away of being knowledgeable in these environments, but I am very stubborn, so It makes up for my lack of knowledge. 

I managed to install psa_car_controller via docker in my windows 11 computer. It all works fine and I find it really nice and useful for my opel corsa-e. I intend to use this to limit my charge at 80%.

My question is simple: do I need to run the package at all times? I happen to have a RPi4 that I use as media server, running 24/7. Should I install psa_car_controller there or can I get away by executing it on my Windows 11 computer whenever I want to change charge limits or charging times? My desktop computer is not always on, but my RPi indeed is.

Thanks a lot for your help.

**Comment by rouke-broersma** [comment]
I am pretty sure the charge limit is sent to the car, you only need to keep psacc running 24/7 if you wish to collect the statistics as well.

**Reply by pauborrasblanco** [reply, marked as answer]
Thanks for sharing this. I have been doing some more testing and finally I decided to install the app in my RPi 4. The process was even easier than on my PC, so I'm super happy with the result.

I found out that this is basically an automation of the android APK (or iPhone app). The only way the current API has to tell the car to stop charging is to delay the charging time while it is charging, just like in the Stellantis app, where it is possible to set a time when you want the car to start charging. The catch, though, is that psa_car_controller can automate when this switch is turned on depending on battery percentage. 

Once the car is disconnected from power, psa_car_controller resets the delayed charge switch to "instant charging", so the next time the car is connected to a power supply it starts charging right away. The switch will only be flipped again (automatically) when the battery reaches the desired % of charge. It is super clever in my opinion, but the limitations come from the API, which is quite crappy tbh.

In short. the app must be running 24/7 to catch all these events, otherwise, the car charges up to 100% the next time it is connected to a charger. I have tried this 3 or 4 times and I can confirm the app needs to be running in a server to work properly.

Hope this helps.

**Reply by drupiler** [reply]
Thank you for this! Was scratching my head as to why my Mokka-e kept charging after I set the percentage threshold to 80% - I ran this on my PC, but obviously it was turned off once I shutdown. I assumed the instruction was sent to the car, but it seems it's not the case. Will try it out through my RP. Thanks once again!
